Add package_python_2_7_10_pybigwig, which will by needed by the next …

dpryan79 opened the pull request:
…deepTools release. Also, python should really produce shared libraries.

The next version of the IUC deepTools package will require pyBigWig, so this PR adds it. Note that this requires linking against libpython, which wasn't possible before because `package_python_2_7` and `package_python_2_7_10` weren't configured to produce it (instead you end up linking against the system libpython...which is just asking for trouble). The `--enable-unicode=ucs4` is required to get python to compile, since otherwise ctypes sometimes tries to use ucs2 while the rest of python uses ucs4 (this is likely a python bug).

It might be good to use additional common options with python's autoconf, such as `--enable-ipv6`, though I don't know of any packages that need that (yet).

mvdbeek commented:
For me this looks reasonable, but can you change the package name to package_python_2_7_10_pybigwig_0_1_9 ?

mvdbeek commented on packages/package_python_2_7_10_pybigwig/tool_dependencies.xml:
```diff
@@ -0,0 +1,30 @@
+                    <package md5sum="8ed6a7c2d03c94ae534f8bc549300f5e">https://pypi.python.org/packages/source/p/pyBigWig/pyBigWig-0.1.9.tar.gz</package>
+                </action>
+                <action type="set_environment">
+                    <environment_variable action="prepend_to" name="PYTHONPATH">$INSTALL_DIR</environment_variable>
```
For libpython you probably also want to prepend to the LD_LIBRARY_PATH, right?

mvdbeek commented:
Also, might it be necessary to do something like [this] (http://koansys.com/tech/building-python-with-enable-shared-in-non-standard-location)?

dpryan79 commented:
Good catch on `LD_LIBRARAY_PATH`, I added `C_INCLUDE_PATH` and `CPLUS_INCLUDE_PATH` as well to the environment variables set by the python packages. I've also renamed the package, it wasn't clear to me what the best practice was there.

The `env.sh` files from the python packages will now prevent runtime linking against the system libpython (at least that worked on my docker-based test environment).

mvdbeek commented:
Good, nice work :+1: . Any thoughts @nsoranzo @bgruening ?

nsoranzo commented:
LGTM

bgruening commented:
Awesome! @dpryan79 still need to fix this matplotlib bug. I guess I will go for a special matplotlib package for the 1.4 version. Will keep you posted.
